jpos/src/main/java/org/jpos/transaction/TransactionManager.java

```diff
@@ -1114,7 +1117,7 @@ public static Long getId() {
     private void setThreadName (long id, String method, TransactionParticipant p) {
         Thread.currentThread().setName(
             String.format("%s:%d %s %s %s", getName(), id, method, p.getClass().getName(),
-                LocalDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()))
+                LocalDateTime.ofInstant(Instant.now(NanoClock.systemUTC()), ZoneId.systemDefault()))
```

Comment:
We don't need NanoClock for monotonic during the living of the same instance of the jvm, `System.nanoTime()` already provides that

Reply:
Well this is epoch based from what I can tell, and `System.nanoTime()` is only a duration by itself.